### PR TITLE
WIP: Terms of Use | API interaction

### PR DESCRIPTION
**Pull request by asg5704**
## Summary
This PR adds the API integration for users to integrate with the Terms of Use actions of Accepting or Declining.
*Note: this will not include the auto-signout after decline button interaction nor redirecting to the `/decline` page

## Testing done
- Unit testing + manual testing

## What areas of the site does it impact?
This will affect just the Terms of Use page `/terms-of-use` in staging & below

### Quality Assurance & Testing

- [x] I added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback
1. Pull down latest in `vets-api` and start the server
2. Pull down this branch `tou/flipper`
3. Run `yarn` then `yarn watch` 
4. Navigate to `localhost:3001/terms-of-use`
    1. Confirm there are no buttons (Accept or Decline) to click because you are not signed in
5. Sign into VA.gov using any available CSP
6. After authenticating navigate to `localhost:3001/terms-of-use`
7. Open up your DevTools on the Network tab
    1.  Click Accept button confirm the API URL includes the `terms_of_use_agreements/v1/accept`
    2. Click Decline button confirm the API URL includes the `terms_of_use_agreeemnts/v1/decline`
8. Stop the `vets-api` server by running Ctrl + C
9. Try to click the Accept/Decline button
10. Confirm there is an error message asking for them to try again.

### CODE REVIEW

File: src/applications/terms-of-use/containers/TermsOfUse.jsx [hunk 1]
```diff
@@ -1,21 +1,78 @@
...
+            message: defaultErrorMessage,
+          });
+        }
+
+        if (Object.keys(response?.termsOfUseAgreement).length) {
+          // if the type was accept
+          if (type === 'accept') {
+            window.location = !redirectUrl
+              ? `${environment.BASE_URL}`
+              : redirectUrl;
```
[the remark this answers is not in the document]
Comment (asg5704) [reply src/applications/terms-of-use/containers/TermsOfUse.jsx:55]: @joeniquette possible XSS here due to the `redirect_url`, let's talk a bit more about this tomorrow.

[the remark this answers is not in the document]
Comment (joeniquette) [reply src/applications/terms-of-use/containers/TermsOfUse.jsx:55]: yea we need to ensure we are only allowing a specific set of base urls, for example the domain must be va.gov.

**Comment by asg5704** [comment]
@department-of-veterans-affairs/va-platform-cop-frontend Can I get a review on this please? Thank you.

**Review by bosawt — APPROVED** [review]
Logged in with id.me user 36, was redirected to terms of use. Then clicked 'agree' and was properly redirected to the expected va.gov auth page. Then logged out, logged back in and was redirected to the va.gov auth page (no terms page since I already accepted)

**Comment by asg5704** [comment]
Closing. This PR, somehow the branches got messed up.

**Pull request closed without merging**